data/transactions/logic/eval.go

```diff
@@ -4026,6 +4135,17 @@ func opAppLocalPut(cx *EvalContext) error {
 		}
 		cx.txn.EvalDelta.LocalDeltas[accountIdx][key] = tv.ToValueDelta()
 	}
+
+	// Enforce maximum value length (also enforced by ledger)
+	if tv.Type == basics.TealBytesType {
```

Comment:
While working on boxes, I noticed that we were not enforcing various constraints in the logic package - only in the ledger. I added these checks here, even though they duplicate ledger checks because I wanted to make it very obvious that Teal/AVM semantics would change if these are changed. We can't just change these values in consensus, we would have to insert a lot of AVM versioning. (For example, suppose we increase the total length allowed for k/v - we would need to enforce that was not visible in previous versions, since the size could then be unexpectedly large when performing a get.  This may have even been a mistake when we introduced values > 64 bytes, and I didn't want to repeat it.)

With these checks in the AVM itself, we could insert some tests, but that's why there are no in package tests - the check didn't used to be here. (And code coverage doesn't notice if a path is taken from unit tests in another package.)

Reply:
@jannotti @algorandskiy I attempted to address the feedback for test coverage in https://github.com/algorand/go-algorand/pull/4693.  Can you review https://github.com/algorand/go-algorand/pull/4693?

I'll resolve the thread once a change is merged in.